cmd/virt-handler/virt-handler.go

```diff
@@ -328,6 +328,9 @@ func (app *virtHandlerApp) Run() {
 		return
 	}
 
+	vmiInformer := factory.VMI()
+	nodeInformer := factory.KubeVirtNode()
```

Comment:
The component which sets topology hints is virt-controller [1], which already has a vmi informer [2].

Does it make sense to move the implementation to virt-controller? IIUC we can simply perform the same logic if the vmi is running, not only pending.

WDYT?

[1] https://github.com/kubevirt/kubevirt/blob/v0.49.1/pkg/virt-controller/watch/vmi.go#L439
[2] https://github.com/kubevirt/kubevirt/blob/v0.49.1/pkg/virt-controller/watch/vmi.go#L221

Reply:
I thought about that too, but then there might be a race between virt-controller and virt-handler as they both have logic around topologyHints, since it's a 0.49 only patch I preferred to do it like this to be on the safe side even if it's a bit more complex.

Reply:
Makes total sense!
Thanks for explaining